pkg/envoy/xds/cache.go

```diff
@@ -208,16 +208,19 @@ func (c *Cache) GetResources(typeURL string, lastVersion uint64, nodeIP string,
 		Canary:  false,
 	}
 
-	// Return all resources.
+	// Return all resources of given typeURL.
 	// TODO: return nil if no changes since the last version?
 	if len(resourceNames) == 0 {
 		res.ResourceNames = make([]string, 0, len(c.resources))
 		res.Resources = make([]proto.Message, 0, len(c.resources))
-		cacheLog.Debugf("no resource names requested, returning all %d resources", len(c.resources))
 		for k, v := range c.resources {
+			if k.typeURL != typeURL {
+				continue
+			}
```

Comment:
What's the user impact of this bug if any? We'll need to understand that to set the proper `release-note` label and write the release note text (in the PR template).

Reply:
returned result of the function will change. Previously the function returned all resources in the cache. After the fix it will return only resources of given typeURL.

Reply:
I understand that aspect, but users don't see the return values of functions, they see changes in behavior. So how will the behavior of Cilium change for users? Or maybe it won't? Basically, how would you describe the impact of this fix to a cluster operator?

Reply:
Hm. Cilium users don't managed any of those resources directly, do they? Is there maybe an impact on Cilium's L7 policies or on the L7 load balancing (Ingress or GatewayAPI)?

Reply:
In NetworkPolicies this function is used only in ``GetNetworkPolicies` which is marked as test only. And all tests passed

Reply:
Ok, so sounds like there's no user-visible impact :+1: